File: Problem84_LoggerRateLimiter.java

// Time Complexity : O(1)
// Space Complexity : O(messgaes)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No


// Your code here along with comments explaining your approach
import java.util.*;
class Logger {
    private HashMap<String, Integer> msgDict;
    public Logger() {
        msgDict = new HashMap<String, Integer>();
    }
    
    public boolean shouldPrintMessage(int timestamp, String message) {
        if (!this.msgDict.containsKey(message)) {
          this.msgDict.put(message, timestamp);
          return true;
        }

        Integer oldTimestamp = this.msgDict.get(message);
        if (timestamp - oldTimestamp >= 10) {
            this.msgDict.put(message, timestamp);
            return true;
        } else
            return false;
    }
}

File: Problem-84 Logger Rate Limiter.py
# 359. Logger Rate Limiter
# https://leetcode.com/problems/logger-rate-limiter/

# Time Complexity: O(1)
# Space Complexity: O(n)

class Logger:
    def __init__(self):
        self.time_dict = dict()

    def shouldPrintMessage(self, timestamp: int, message: str) -> bool:
        if message not in self.time_dict:
            self.time_dict[message] = timestamp + 10
            return True
        else:
            if self.time_dict[message] <= timestamp:
                self.time_dict[message] = timestamp + 10
                return True
            else:
                return False


# Your Logger object will be instantiated and called as such:
# obj = Logger()
# param_1 = obj.shouldPrintMessage(timestamp,message)